yarn-project/aztec-rpc/src/simulator_oracle/index.ts

```diff
@@ -38,13 +39,7 @@ export class SimulatorOracle implements DBOracle {
    * @throws An Error if the input address does not match the public key address of the key pair.
    */
   getSecretKey(_contractAddress: AztecAddress, pubKey: Point): Promise<Buffer> {
-    const thisPubKey = this.keyPair.getPublicKey();
-    if (!thisPubKey.equals(pubKey)) {
-      throw new Error(
-        `Only allow access to the secret keys of the tx creator (requested keys for ${pubKey}, expected ${thisPubKey}).`,
-      );
-    }
-    return this.keyPair.getPrivateKey();
+    return this.keyStore.getAccountPrivateKey(pubKey);
```

Comment:
Should we create an issue for restricting which keys can be accessed during a simulation, based on accesses granted by the user?

Reply:
#1068